libraries/AP_HAL/I2CDevice.h

```diff
@@ -27,6 +27,8 @@ namespace AP_HAL {
 
 class I2CDevice : public Device {
 public:
+    I2CDevice() : Device(I2C) { }
```

Comment:
That was my first approach. See https://github.com/ArduPilot/ardupilot/pull/4366#commitcomment-17990056 .